docs/en/observability/monitor-java-app.asciidoc

```diff
@@ -0,0 +1,2005 @@
+}
+
+test {
+  useJUnitPlatform()
+}
+----
++
+. Run the following command.
++
+[source,bash]
+----
+echo "rootProject.name = 'javalin-app'" >> settings.gradle
+
+mkdir -p src/main/java/de/spinscale/javalin
+mkdir -p src/test/java/de/spinscale/javalin
+----
++
+. Install the Gradle wrapper. An easy way to install Gradle is to use
+https://sdkman.io/[sdkman] and run `sdk install gradle 6.5.1`.
++
+. Run `./gradlew clean check`. You should see a
+successful build that has nothing built or compiled yet.
++
+. To create a Javalin server and its first endpoint (the main endpoint), create the
+`src/main/java/de/spinscale/javalin/App.java` file.
++
+[source,java]
+----
+package de.spinscale.javalin;
+
+import io.javalin.Javalin;
+
+public class App {
+    public static void main(String[] args) {
+        Javalin app = Javalin.create().start(7000);
+        app.get("/", ctx -> ctx.result("Appsolutely perfect"));
+    }
+}
+----
++
+. Run `./gradlew assemble`
++
+This command compiled the `App.class` file in the `build` directory.
+However, there is no way to start the server. Let’s create a jar which
+contains our compiled class along with all the required dependencies.
++
+. In the `build.gradle` file, edit `plugins` as shown here.
++
+[source,gradle]
+----
+plugins {
+  id 'com.github.johnrengelman.shadow' version '6.0.0'
+  id 'java'
+}
+----
++
+. Run `./gradlew shadowJar`. This command creates a
+`build/libs/javalin-app-all.jar` file.
++
+The `shadowJar` plugin requires information about its main class.
+. Add the following snippet to the `build.gradle` file.
++
+[source,gradle]
+----
+jar {
+  manifest {
+    attributes 'Main-Class': 'de.spinscale.javalin.App'
+  }
+}
+----
++
+. Rebuild the project and start the server.
++
+[source,bash]
+----
+java -jar build/libs/javalin-app-all.jar
+----
++
+Opening another terminal and running
+`curl localhost:7000` displays an HTTP response.
++
+. Test the code. Placing everything into the `main()` method makes it difficult to test
+the code. However, a dedicated handler fixes this.
++
+Refactor the `App` class.
++
+[source,java]
+----
+package de.spinscale.javalin;
+
+import io.javalin.Javalin;
+import io.javalin.http.Handler;
+
+public class App {
+    
+    public static void main(String[] args) {
+        Javalin app = Javalin.create().start(7000);
+        app.get("/", mainHandler());
+    }
+
+    static Handler mainHandler() {
+        return ctx -> ctx.result("Appsolutely perfect");
+    }
+}
+----
++
+Add a Mockito and an Assertj dependency to the `build.gradle` file.
++
+[source,gradle]
+----
+dependencies {
+  implementation 'io.javalin:javalin:3.9.1'
+
+  testImplementation 'org.mockito:mockito-core:3.3.3'
+  testImplementation 'org.assertj:assertj-core:3.16.1'
+  testImplementation 'org.junit.jupiter:junit-jupiter-api:5.6.2'
+  testRuntimeOnly 'org.junit.jupiter:junit-jupiter-engine:5.6.2'
+}
+----
++
+Create an `AppTests` class in `src/test/java/de/spinscale/javalin`.
```

Comment:
I can't get to this step, but it isn't clear to me that this should be in a file named `AppTests.java`. We should clarify.

Reply:
++